sda-commons-server-kafka/src/test/java/org/sdase/commons/server/kafka/KafkaTopicIT.java

```diff
@@ -117,28 +118,31 @@ public void checkTopicSuccessfulComplex() {
     assertThat(stringStringMessageListener).isNotNull();
   }
 
-  @Test(expected = MismatchedTopicConfigException.class)
-  public void checkTopicFails() {
-
-    List<MessageListener<String, String>> stringStringMessageListener =
-        bundle.createMessageListener(
-            MessageListenerRegistration.builder()
-                .withDefaultListenerConfig()
-                .forTopic("SomeNotExisting")
-                .checkTopicConfiguration()
-                .withDefaultConsumer()
-                .withValueDeserializer(new StringDeserializer())
-                .withListenerStrategy(
-                    new AutocommitMLS<String, String>(
-                        record -> results.add(record.value()),
-                        new IgnoreAndProceedErrorHandler<>()))
-                .build());
-
-    assertThat(stringStringMessageListener).isNotNull();
+  @Test
+  void checkTopicFails() {
+    assertThatCode(
+            () -> {
+              List<MessageListener<String, String>> stringStringMessageListener =
+                  bundle.createMessageListener(
+                      MessageListenerRegistration.builder()
+                          .withDefaultListenerConfig()
+                          .forTopic("SomeNotExisting")
+                          .checkTopicConfiguration()
+                          .withDefaultConsumer()
+                          .withValueDeserializer(new StringDeserializer())
+                          .withListenerStrategy(
+                              new AutocommitMLS<String, String>(
+                                  record -> results.add(record.value()),
+                                  new IgnoreAndProceedErrorHandler<>()))
+                          .build());
+
+              assertThat(stringStringMessageListener).isNotNull();
+            })
+        .isInstanceOf(MismatchedTopicConfigException.class);
```

Comment:
Is the assert statement in line 139 really executed? You are expecting the previous code to throw, so I'm a bit confused.

Reply:
I agree and expect that line 139 will not be executed. But actually line 137 in the test before did not either. I disagree with Sonar although it catches the same point here. If you follow Sonars advice completely you need to pick the exact method that should throw the exception but in my opinion we should ensure that building a message listener with this configuration throws an exception in total and not a single specific method. It is an integration test. That check was added in May last year and since then it's annoying me from time to time because tests often get less readable when you follow that rule.

Reply:
I'll remove line 139 as it is not needed. Didn't see that when converting the test from JUnit 4 to 5.